crates/symbolicator/src/services/bitcode.rs

```diff
@@ -262,31 +262,19 @@ impl BitcodeService {
         scope: Scope,
         sources: Arc<[SourceConfig]>,
     ) -> Result<Option<Arc<CacheHandle>>, Error> {
-        sentry::with_scope(
-            |scope| {
-                scope.set_tag("auxdif.debugid", uuid);
-                scope.set_extra("auxdif.kind", dif_kind.to_string().into());
-            },
-            || async {
-                let mut jobs = Vec::with_capacity(sources.len());
-                for source in sources.iter() {
-                    let job =
-                        self.fetch_file_from_source(uuid, dif_kind, scope.clone(), source.clone());
-                    jobs.push(job);
-                }
-                let results = future::join_all(jobs).await;
-                let mut ret = None;
-                for result in results {
-                    match result {
-                        Ok(Some(handle)) => ret = Some(handle),
-                        Ok(None) => (),
-                        Err(err) => return Err(err),
-                    }
-                }
-                Ok(ret)
-            },
-        )
-        .await
+        let jobs = sources.iter().map(|source| {
+            self.fetch_file_from_source(uuid, dif_kind, scope.clone(), source.clone())
+        });
+        let results = future::join_all(jobs).await;
```

Comment:
So I think the premise of this PR is a little wrong and whenever you do go into a number of futures which are running concurrently, like with `future::join_all`, you need to give them each a new hub.  The reason is that they call can be pushing and popping scopes, change tags and context on scopes etc.  Since all the futures run concurrently they would be influencing each other a little too much.

So while not a regression here, when creating the jobs here they should all have received a new hub.